### PR TITLE
Fixes for indices in the mempool

### PR DESCRIPTION
**Pull request by domob1812**
This set of changes fixes how the mempool tracks the spent and address index, in particular when transactions get removed (e.g. because they are confirmed now).  Previously, the mempool was never actually removing any of the index data, which lead to incorrect results (the new regtest fails without both the fixes for the address and spent index).

### CODE REVIEW

File: divi/src/txmempool.cpp [hunk 1]
```diff
@@ -23,6 +23,9 @@
 
 using namespace std;
 
+extern bool fAddressIndex;
+extern bool fSpentIndex;
+
```
Comment (galpHub) [inline divi/src/txmempool.cpp]: Might be preferable to inject these as const& dependencies  of the mempool class. The flags are already defined adjacent to the definition of the global mempool so it's not that much of a stretch. This also makes the dependency of the mempool on these a little more explicit in the instantiation.

Comment (domob1812) [reply divi/src/txmempool.cpp]: Sounds good, I'll update the PR accordingly.

**Comment by galpHub** [comment]
Looks good 👍

**Comment by domob1812** [comment]
Thanks for the review, updated.

**Pull request merged**